eng/pipelines/variables/common.yml

```diff
@@ -10,14 +10,25 @@ variables:
   value: false
 - name: publicGitRepoUri
   value: https://github.com/dotnet/dotnet-buildtools-prereqs-docker
-- name: manifestVariables
-  value: --var UniqueId=$(sourceBuildId)
-- name: publicSourceBranch
-  value: main
+- ${{ if eq(variables['Build.SourceBranch'], 'refs/heads/production') }}:
+  - name: manifestVariables
+    value: --var UniqueId=$(sourceBuildId) --var FloatingTagSuffix=latest
```

Comment:
One thing you may consider is refactoring if else to set a new floating tag suffix variable and then define the `manifestVariables` outside the if else.  The primary benefit it to ensure consistency and avoid duplication.

```
- ${{ if eq(variables['Build.SourceBranch'], 'refs/heads/production') }}:
  - name: floatingTagSuffix
    value: latest
- ${{ elseif eq(variables['Build.SourceBranch'], 'refs/heads/main') }}:
  - name: floatingTagSuffix
    value: staging
- ${{ else }}:
  - name: floatingTagSuffix
    value: $(Build.SourceBranchName)
- name: manifestVariables
  value: --var UniqueId=$(sourceBuildId)  --var FloatingTagSuffix=$(floatingTagSuffix)
```

Reply:
I like that. Updating.